src/main/java/org/zaproxy/admin/GenerateAddonsYAML.java

```diff
@@ -0,0 +1,108 @@
+ *
+ * Copyright 2015 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.admin;
+
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.nio.file.Files;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Command line tool for generating the addons markdown file for website.
+ *
+ * @author nirojan
+ */
+public class GenerateAddonsYAML {
+
+    private static final String ZAP_VERSIONS_FILE_NAME = "ZapVersions-2.9.xml";
+    private static final String NEXT_LINE = "\n";
+    private static final String EMPTY_STRING = "";
+    private static final String OUTPUT_DIR = "addons.yml";
+
+
+    public static void main(String[] args) throws Exception {
+
+        StringBuilder sb = new StringBuilder("---" + NEXT_LINE);
+        File xmlFile = new File(ZAP_VERSIONS_FILE_NAME);
+        if(xmlFile.exists()){
+
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            org.w3c.dom.Document doc = builder.parse(ZAP_VERSIONS_FILE_NAME);
+
+            NodeList list = doc.getElementsByTagName("addon");
+
+            for (int i=0; i<list.getLength(); i++) {
+
+                Element addonParentElement = (Element)list.item(i);
+                NodeList addonIndividualNode = doc.getElementsByTagName("addon_" + addonParentElement.getTextContent());
+
+                if(addonIndividualNode.getLength() != 0){
+                    Element addonInfo = (Element)addonIndividualNode.item(0);
+
+                    String addonName = getAddonDetails("name", addonInfo);
+                    String description = getAddonDetails("description", addonInfo);
+                    String author = getAddonDetails("author", addonInfo);
+                    String version = getAddonDetails("version", addonInfo);
+                    String file = getAddonDetails("file", addonInfo);
+                    String status = getAddonDetails("status", addonInfo);
+                    String url = getAddonDetails("url", addonInfo);
+                    String date = getAddonDetails("date", addonInfo);
+                    String size = getAddonDetails("size", addonInfo);
+
+                    sb.append("- name: " + addonName + NEXT_LINE)
```

Comment:
We should also include the ID of the add-on (that was added recently to the site to allow to link directly to each add-on) and the repo URL.